mozilla/blurts-server
# send localized breach alerts

# References: 
Jira: MNTOR-2120




# Description

The previous localization method wasn't working as expected because it was using [AsyncLocalStorage](https://nodejs.org/api/async_context.html#class-asynclocalstorage), which requires a chain of asynchronous operations. This worked fine when this was all running under an HTTP request (Express, and then tested under Cloud Functions), but as a standalone Node cron job this doesn't work as expected (`getStore()` always returns `undefined`).

Note: I noticed this when testing against stage, the feature flag allowing new breaches on prod is currently paused, I'd like to get this pushed to prod before we re-enable.

# How to test

- create an account in your local instance using browser locale set to something other than English (stored as `signup_language` in `subscribers` table)
- have a known `sha1` hash of some test email addresses in your local database, for instance `test@example.com` is `567159d622ffbb50b11b0efd307be358624a26ee` (but it should be an email address you can check the inbox of)
- run the Pub/Sub emulator per the README
- `curl -v -d '{ "breachName": "000webhost", "hashPrefix": "56715", "hashSuffixes": ["9d622ffbb50b11b0efd307be358624a26ee"] }' -H "Authorization: Bearer ${HIBP_NOTIFY_TOKEN}" http://localhost:6060/api/v1/hibp/notify`
- `node src/scripts/emailBreachAlerts.js` <- this will email `test@example.com`

# Checklist (Definition of Done)
